lib/migrations/state.js

```diff
@@ -0,0 +1,117 @@
+/**
+ * migrations/state.js - migration state for hsd.
+ * Copyright (c) 2021, Nodari Chkuaselidze (MIT License)
+ */
+'use strict';
+
+const assert = require('bsert');
+const bio = require('bufio');
+const {encoding} = bio;
+
+/**
+ * State of database migrations.
+ * because migration IDs are only increasing, we only need
+ * to store the last one.
+ * @alias module.migrations.MigrationState
+ * @property {Boolean} inProgress - is/was migration in progress
+ * NOTE: If inProgress is true, we know lastMigration + 1 was the one in
+ *       the progress.
+ * @property {Number} lastMigration - last migration
+ * NOTE: Migration numbers start from 1.
+ *       0 - means there is no migration.
+ */
+
+class MigrationState extends bio.Struct {
+  /**
+   * Create MigrationState
+   * @constructor
+   */
+
+  constructor() {
+    super();
+
+    this.inProgress = false;
+    this.nextMigration = 0;
+    this.skipped = [];
+  }
+
+  get lastMigration() {
+    return this.nextMigration - 1;
+  }
+
+  /**
+   * Inject properties from another state.
+   * @param {MigrationState} obj
+   * @returns MigrationState;
+   */
+
+  inject(obj) {
+    assert(obj instanceof MigrationState);
+    this.inProgress = obj.inProgress;
+    this.nextMigration = obj.nextMigration;
+    this.skipped = obj.skipped.slice();
+    return this;
+  }
```

Comment:
Is this ever used? Or just to have general compatibility with `bio.Struct` ?

Reply:
This one is not used anymore, it was for better `.clone`. But left it there if we need to clone it again.

Reply:
ok :+1: